src/Oscoin/Crypto/Address/Internal.hs

```diff
@@ -80,14 +80,14 @@ data AddressFormatTag =
     AddressFormatTag_CBOR
     deriving (Eq, Ord, Show)
 
--- | The 'AddressPayload' is (currently) simply a 'PublicKey'.
+-- | The 'AddressPayload' is a 'ShortHash' of a 'PublicKey'.
 --
 data AddressPayload c where
-    AddressPayload_V0 :: PublicKey c -> AddressPayload c
+    AddressPayload_V0 :: ShortHash c -> AddressPayload c
```

Comment:
In my very first iteration of the address format, I wrote [this](https://github.com/oscoin/oscoin/pull/431/files#diff-0a67e40a5b386d32bba504d153d37813R92). How did you avoid this problem? I guess you did use the `parseXXX` machinery to do that, but I do wonder if it might be easier simply to have a `ByteArray` there, if that clean up things. Not proposing to actually go with this design, but just flagging up that might be a route worth exploring 😉

Reply:
I also wondered - I don't quite understand what the stated problem was to be honest, due to my limited haskell experience :confused: 

Could it be that I used `ShortHash` instead of `ShortHashed`?

Reply:
(Going by memory here): Back in the days I think the problem was that I wanted to use a `ShortHashed` to be able to "remember" which was the thing being hashed (perhaps I thought it was useful for migration purposes, who knows). But that wasn't possible (for the reasons I have explained in the comment, essentially there was no inverse function during decoding to go from a `ShortHashed` back to the `PublicKey`) so the only way to encode the fact that Hash was actually hashing a `PublicKey` was using `Tagged`. Perhaps it's useful to do the same here?

Reply:
Right, that's what I was thinking. I decided against using `Hashed` for exactly this reason - since we don't always have the original `PublicKey`, and would have to "cheat" it. I think this is the case almost always, and so I think `Hashed` is actually not that useful and should potentially be removed.